rskj-core/src/main/java/co/rsk/rpc/netty/ExecTimeoutContext.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2018 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.rpc.netty;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class ExecTimeoutContext implements AutoCloseable {
+
+    private static final ThreadLocal<Set<ExecTimeoutContext>> sExecTimeoutContext = ThreadLocal.withInitial(HashSet::new);
```

Comment:
Hey @rmoreliovlabs, @Vovchyk, first of all I liked the idea of the `ExecTimeoutContext` 👍 

One question though: Could you please tell me why we need a set here? Couldn’t we just create one `ExecTimeoutContext` in `JsonRpcCustomServer`? Probably I'm missing sth but preferred to ask.

Reply:
@illuque-iov , generally speaking there could be a [batch](https://geth.ethereum.org/docs/interacting-with-geth/rpc/batch) request being processed with arbitrary hierarchy, eg. [[eth_call(), eth_chainId()], [eth_gatLogs]]. So that besides a root timeout we may need to handle rpc method level timeout (kinda nested timeout). A `Set<>` here allows us to created more than one context for any particular thread, if needed

Reply:
Oh, I did not know about that batch possibility. Thanks guys and good job 😉.